pytest_operator/plugin.py

```diff
@@ -342,46 +343,143 @@ def download_resource(self, resource, destination: Path):
         return local_file
 
 
+ModelKey = namedtuple("ModelKey", "controller, cloud, model")
+
+
+class ModelNotFoundError(Exception):
+    """Raise when switching to a model that doesn't exist."""
+
+
+@dataclasses.dataclass
+class ModelState:
+    model: Model
+    tmp_path: Optional[Path]
```

Comment:
each model will get a new temporary path if it wishes to render bundles, download resources, build charms -- etc